source/module_hamilt_lcao/module_tddft/td_velocity.cpp

```diff
@@ -0,0 +1,130 @@
+{}
+
+void TD_Velocity::output_cart_At(const std::string &out_dir)
+{
+  if(GlobalV::MY_RANK==0)
+  {
+    std::string out_file;
+    //generate the output file name
+    out_file= out_dir + "At.dat";
+    std::ofstream ofs;
+    //output title
+    if(istep==0)
+    {
+      ofs.open(out_file.c_str(), std::ofstream::out);
+      ofs << std::left 
+          << std::setw(8) << "#istep"
+          << std::setw(15) << "A_x" 
+          << std::setw(15) << "A_y" 
+          << std::setw(15) << "A_z" << std::endl;
+    }
+    else
+    {
+      ofs.open(out_file.c_str(), std::ofstream::app);
+    }
+    //output the vector potential
+    ofs << std::left <<std::setw(8) << istep;
+    // divide by 2.0 to get the atomic unit
+    for(int i=0; i<3; i++)
+      ofs << std::scientific << std::setprecision(4)<< std::setw(15) << cart_At[i]/2.0;
+    ofs << std::endl;
+    ofs.close();
+  }
+  return;
+}
+
+void TD_Velocity::cal_cart_At(ModuleBase::Vector3<double> a0, 
+                              ModuleBase::Vector3<double> a1, 
+                              ModuleBase::Vector3<double> a2,
+                              ModuleBase::Vector3<double> At)
+{
+  istep++;
+  if(init_vecpot_file)
+  {
+    this->cart_At=At_from_file[istep>max_istep?max_istep:istep];
+  }
+  else
+  {
+    double l_norm[3]={a0.norm() ,a1.norm() ,a2.norm()};
+	  this->cart_At = a0*At[0]/l_norm[0] + a1*At[1]/l_norm[1] + a2*At[2]/l_norm[2];
+  }
+  //output the vector potential if needed
+  if(out_vecpot==true)
+  {
+    this->output_cart_At(GlobalV::global_out_dir);
+  }
+}
+
+void TD_Velocity::read_cart_At(void)
+{
+
+  std::string in_file;
+  //generate the input file name
+  in_file= "At.dat";
+  std::ifstream ifs(in_file.c_str());
+  //check if the file is exist
+  if(!ifs)
+  {
+    ModuleBase::WARNING_QUIT("TD_Velocity::read_cart_At", "Cannot open Vector potential file!");
+  }
+  std::string line;
+  std::vector<std::string> str_vec;
+  //use tmp to skip the istep number
+  int tmp=0;
+  while(std::getline(ifs, line))
+  {
+    //A tmporary vector3 to store the data of this line
+    ModuleBase::Vector3<double> At;
+    if(line[0]=='#')
+      continue;
+    std::istringstream iss(line);
+    //skip the istep number
+    if (!(iss >> tmp)) {
+      ModuleBase::WARNING_QUIT("TD_Velocity::read_cart_At", "Error reading istep!");
+    }
+    //read the vector potential
+    double component;
+    // Read three components
+    for(int i = 0; i < 3; i++)
+    {
+      if (!(iss >> component)) 
+      {
+        ModuleBase::WARNING_QUIT("TD_Velocity::read_cart_At", "Error reading component " + std::to_string(i+1) + " for istep " + std::to_string(tmp) + "!");
+      }
+      At[i] = component;
+    }
+    //unit transform ,change unit into Ry/bohr/e*t_a.u.
+    At*=2.0;
+    //add the tmporary vector3 to the vector potential vector
+    At_from_file.push_back(At);
+  }
+  //set the max_istep
+  max_istep = At_from_file.size()-1;
+  ifs.close();
+
+  return;
+}
```

Comment:
Did you use 'clang_format' for all of the files?

Reply:
Sorry, I don't know what it is. Could you please tell me more about it? @mohanchen